src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -246,7 +245,7 @@ protected function addFieldToOperationSettings($field)
         $fieldKey = $this->getFieldKey($field);
 
         $allFields = $this->getOperationSetting('fields');
-        $allFields = Arr::add($this->fields(), $fieldKey, $field);
+        $allFields = array_merge($this->fields(), [$fieldKey => $field]);
```

Comment:
`Arr::add` here would parse relation names like `address.street` and create a multi-dimensional array from it `address => [street => ]` and what we want is to add the full field key `address.street` without dot notation parsing